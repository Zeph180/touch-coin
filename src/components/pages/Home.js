import { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { Icon } from '@iconify/react';
import { getCryptoCoinsAsync } from '../redux/crypto/cryptoCoinSlice';
import '../styles/Home.css';

export default function Home() {
  const dispatch = useDispatch();
  const [searchQuery, setSearchQuery] = useState('');

  useEffect(() => {
    dispatch(getCryptoCoinsAsync());
  }, [dispatch]);

  const { cryptoCoins } = useSelector((state) => state.cryptoCoins);

  if (!cryptoCoins) {
    return <p>No coins loaded</p>;
  }

  const filteredCoins = cryptoCoins
    .filter((coin) => coin.name.toLowerCase()
      .includes(searchQuery.toLowerCase()));

  return (
    <>
      <div className="nav">
        <h1>Touch coin</h1>
        <div className="icons-cont">
          <Icon icon="mdi:microphone" />
          <Icon icon="material-symbols:settings" />
        </div>
      </div>
      <div className="search-bar">
        <input
          type="text"
          placeholder="Search coins"
          value={searchQuery}
          onChange={(e) => setSearchQuery(e.target.value)}
        />
      </div>
      <section className="home-cont">
        {filteredCoins.map((cryptoCoin) => (
          <article className="coin" key={cryptoCoin.id}>
            <Link to={`touch-coin/details/${cryptoCoin.id}`}>
              <Icon className="link" icon="mdi:arrow-right-thin-circle-outline" />
            </Link>
            <article className="logo-cont">
              <div className="logo">
                <img
                  src={cryptoCoin.icon}
                  alt={`${cryptoCoin.name}logo`}
                  className="icon"
                />
              </div>
            </article>
            <article className="brief">
              <h2 className="heading">{cryptoCoin.name}</h2>
              <p>{cryptoCoin.totalSupply}</p>
            </article>
          </article>
        ))}
      </section>
    </>
  );
}
