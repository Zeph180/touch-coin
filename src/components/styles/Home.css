body {
  margin: 0;
  color: white;
}

h1 {
  margin: 0;
}

.nav {
  background-color: #a30559;
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
}

.icons-cont {
  width: 15%;
  font-size: 25px;
  display: flex;
  justify-content: space-between;
}

.home-cont {
  display: grid;
  overflow: hidden;
  grid-template-columns: 50% 50%;
}

.home-cont > :nth-child(4n+2),
.home-cont > :nth-child(4n+3) {
  background-color: #ff69b4;
}

.coin {
  cursor: pointer;
  height: 180px;
  background-color: #ff1493;
}

.coin:hover {
  background-color: #6e083f;
}

.link {
  position: absolute;
  margin: 10px 0 0 40%;
  color: #fff;
  font-size: 25px;
}

.logo-cont {
  height: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.logo {
  margin-top: 25px;
  width: 90px;
  height: 90px;
  border-radius: 50px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.icon {
  height: 100%;
  width: 100%;
}

.brief {
  text-align: right;
  margin: 20px 10px 0 0;
}

.heading {
  margin: 20px 0 5px;
}

p {
  margin: 0;
}
