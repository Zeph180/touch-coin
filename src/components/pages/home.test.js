import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/app/Store';
import Home from './Home';

describe('Home component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
