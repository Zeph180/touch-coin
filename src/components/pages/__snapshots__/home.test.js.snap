// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav"
  >
    <h1>
      Touch coin
    </h1>
    <div
      class="icons-cont"
    >
      <span />
      <span />
    </div>
  </div>
  <div
    class="search-bar"
  >
    <input
      placeholder="Search coins"
      type="text"
      value=""
    />
  </div>
  <section
    class="home-cont"
  />
</DocumentFragment>
`;
